Fill.py

```diff
@@ -945,18 +961,17 @@ def failed(warning: str, force: typing.Union[bool, str]) -> None:
                 failed(
                     f"Plando block failed to place {m - count} of {m} item(s) for {world.player_name[player]}, error(s): {' '.join(err)}",
                     placement['force'])
-            for (item, location) in successful_pairs:
+
+            # Sort indices in reverse so we can remove them one by one
+            successful_pairs.sort(key=lambda index_and_placement: index_and_placement[0], reverse=True)
+
+            for (index, item, location) in successful_pairs:
                 world.push_item(location, item, collect=False)
                 location.event = True  # flag location to be checked during fill
                 location.locked = True
                 logging.debug(f"Plando placed {item} at {location}")
-                if from_pool:
-                    try:
-                        world.itempool.remove(item)
-                    except ValueError:
-                        warn(
-                            f"Could not remove {item} from pool for {world.player_name[player]} as it's already missing from it.",
-                            placement['force'])
+                if index is not None:
```

Comment:
just pop this in the try?

Reply:
If this is a reply to the "two O(n)" thing: That's still next, followed by pop. Not an improvement

Reply:
It makes the index thing less awkward / obesolete, tho, so it's probably better

Reply:
it removes the if check, so it's still marginally faster

Reply:
Yeah, makes the code shorter as well.

My new solution depends on whether you hate "for - else"